testsuite/testsuite-keycloak/src/test/java/org/wildfly/swarm/keycloak/KeycloakArquillianTest.java

```diff
@@ -15,40 +15,122 @@
+        deployment.addResource(SecuredApplication.class);
+        deployment.addResource(SecuredResource.class);
+        deployment.addAsResource("wildfly-swarm-keycloak-example-realm.json");
+        deployment.addAsResource("keycloak.json");
         return deployment;
     }
 
     @CreateSwarm
     public static Swarm newContainer() throws Exception {
-        return new Swarm().fraction(new KeycloakFraction());
+        String warName = getWarName();
+        File file = File.createTempFile("project-defaults", ".yaml");
+        file.deleteOnExit();
+        createSwarmConfiguration(file, warName);
+
+        URL keyCloakJsonUrl = KeycloakArquillianTest.class.getResource("/keycloak.json");
+        System.setProperty("swarm.keycloak.json.path", keyCloakJsonUrl.getPath());
+        URL migrationRealmUrl = KeycloakArquillianTest.class.getResource("/wildfly-swarm-keycloak-example-realm.json");
+        System.setProperty("keycloak.migration.file", migrationRealmUrl.getPath());
+        System.setProperty("keycloak.migration.provider", "singleFile");
+        System.setProperty("keycloak.migration.action", "import");
+        return new Swarm("-s" + file.toURI().toURL().toString());
     }
 
     @Test
-    @RunAsClient
-    public void testNothing() throws InterruptedException {
-        // TODO figure out how to confirm keycloak is working.
+    public void testResourceIsSecured() throws Exception {
+        // Check 401 is returned without the token
+        Assert.assertEquals(Response.Status.UNAUTHORIZED.getStatusCode(),
+           ClientBuilder.newClient().target("http://localhost:8080/secured").request().get().getStatus());
+        
+        final String tokenUri = 
+            "http://localhost:8080/auth/realms/wildfly-swarm-keycloak-example/protocol/openid-connect/token";
+        String response = 
+            ClientBuilder.newClient().target(tokenUri).request()
+                .post(Entity.form(
+                        new Form().param("grant_type", "password").param("client_id", "curl")
+                                  .param("username", "user1").param("password", "password1")),
+                        String.class);
+        String accessToken = getAccessTokenFromResponse(response);
+        
+        String serviceResponse = 
+            ClientBuilder.newClient().target("http://localhost:8080/secured")
+                .request()
+                .header(HttpHeaders.AUTHORIZATION, "Bearer " + accessToken)
+                .get(String.class);
+        Assert.assertEquals("This resource is secured", serviceResponse);
+
     }
 
+    private String getAccessTokenFromResponse(String response) {
+        String tokenStart = response.substring("{\"access_token\":\"".length());
+        return tokenStart.substring(0, tokenStart.indexOf("\""));
+    }
+
+    private static String getWarName() throws Exception {
+        // Arquillian creates a war using a "wfswarm" + UUID + ".war" format.
```

Comment:
I'm not sure if it would work with Swarm Arquillian adapter, but you can provide an archive name in the `ShrinkWrap.create()` method call.

Reply:
No, that does not work, it still uses UUID, I tried it few times earlier

Reply:
Ah OK, sorry. We should probably fix it :-D

Reply:
Hi Ladislav, np at all. Interesting, is it in our control then :-) ? Yes, see some Arquillian related code...I can definitely open an issue to track it ?